lua/lspconfig/groovyls.lua

```diff
@@ -12,7 +12,10 @@ configs[name] = {
       bin_name,
     },
     filetypes = { 'groovy' },
-    root_dir = util.find_git_ancestor or vim.loop.os_homedir(),
+    root_dir = function(fname)
+      local root_files = { 'Jenkinsfile' }
+      return util.root_pattern(unpack(root_files))(fname) or util.find_git_ancestor(fname)
```

Comment:
Just inline `{ 'Jenkinsfile' }` imo.

Reply:
Without the table ;)

You can probably further compress it as `util.root_pattern('Jenkinsfile') or util.find_git_ancestor`?

Reply:
Yep, I figured a table in case people named their Jenkins file `jenkinsfile` in all lower-case, however the correct filename is `Jenkinsfile` with upper case `J`, so fine with this.